pkg/pgmodel/migrations/sql/1_base_schema.up.sql

```diff
@@ -171,47 +182,91 @@ $func$
 LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_default_retention_period() TO prom_reader;
 
+--lock-order: metric table, data_table, series parent, series partition
+CREATE PROCEDURE SCHEMA_CATALOG.finalize_metric_creation()
```

Comment:
Add a comment explaining why we're splitting these steps out
Also, explain how this is run in the comment, but that we build it to allow concurrency